docs/design/jet/022-mongodb-connector.md

```diff
@@ -0,0 +1,83 @@
+---
+title: 022 - Low code / no code MongoDB connector
+description: Describes the design of low/no code connector for MongoDB 
+---
+
+MongoDB is a widely used NoSQL, document-based database. Although there
+was a Jet sink and source connector implemented in 
+the `hazelcast-jet-contrib` repository, users didn't really have 
+properly tested and supported connector for this database.
+
+Such connector should:
+ - have minimal API available to quickly prototype and more advanced
+   one for those who need many customizations
+ - support parallelism where possible
+ - support encryption for data-in-transit
+ - support SQL <-> MongoDB
+ - support no/low code `MapStore` feature
+ - be tested with integration and soak tests
+
+## Non-goals
+
+It's not a goal of this connector to control MongoDB settings; any 
+changes in MongoDB configuration or any administration action will still 
+be needed to be done on MongoDB side.
+
+
+## Implementation details
+
+### Source and sink
+The `hazelcast-jet-contrib` repository contained MongoDB connector. It was
+based on the `SourceBuilder`, which is nice in general, but for more 
+flexibility we need to rework this connector to low-level Processor/
+ProcessorSupplier/ProcessorMetaSupplier. 
```

Comment:
Clarify why we need this. (I suppose it is the loading of the replica set details?)

Reply:
This + to be able to use it in SQL Connector